pyaerocom/colocation/colocated_data.py

```diff
@@ -128,13 +128,11 @@ def validate_data(self):
             # make sure path is str instance
             self.data = str(self.data)
         if isinstance(self.data, str):
-            assert self.data.endswith("nc"), ValueError(
-                "Invalid data filepath str, must point to a .nc file"
-            )
+            if not self.data.endswith("nc"):
+                raise ValueError("Invalid data filepath str, must point to a .nc file")
             self.open(self.data)
```

Comment:
It is currently not clear that read files are dimension validated without going two function calls deep, which makes this look like a bug without being one. Maybe it can be made clearer?